format/avro/codecs/long.go

```diff
@@ -0,0 +1,34 @@
+package codecs
+
+import (
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+const intMask = byte(127)
+const intFlag = byte(128)
+
+// readLong reads a variable length zig zag long from the current position in decoder
+func VarZigZag(d *decode.D) int64 {
+	var value uint64
+	var shift uint
+	for d.NotEnd() {
+		b := byte(d.U8())
+		value |= uint64(b&intMask) << shift
+		if b&intFlag == 0 {
+			return int64(value>>1) ^ -int64(value&1)
```

Comment:
Its mostly the same, except this can be variable length number of bytes. Same as what was used in the protobuf decoder, except without revealing the intermediate value.

https://avro.apache.org/docs/current/spec.html#binary_encoding